components/constellation/constellation.rs

```diff
@@ -1922,49 +1926,64 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
-    fn traverse_frame_to_pipeline(&mut self, frame_id: FrameId, next_pipeline_id: PipelineId) {
+    fn traverse_frame_to_timestamp(&mut self, frame_id: FrameId, instant: Instant) {
         // Check if the currently focused pipeline is the pipeline being replaced
         // (or a child of it). This has to be done here, before the current
         // frame tree is modified below.
         let update_focus_pipeline = self.focused_pipeline_in_tree(frame_id);
 
-        let prev_pipeline_id = match self.frames.get_mut(&frame_id) {
+        let (prev_pipeline_id, next_pipeline_id, entry) = match self.frames.get_mut(&frame_id) {
```

Comment:
Could we have this match return the previous entry and the next entry instead of returning the pipelines? This will be needed in the future and you're already returning both `next_pipeline_id` and `entry`.